### PR TITLE
BasicDataTypesTestSuite reworked

### LINKED ISSUE #111: Refactor: Rework BasicDataTypesTestSuite be more in-line with newer tests
### Description

BasicDataTypesTestSuite contains 5 facts at the moment, with multiple asserts per fact. This should be broken up into more facts to be more thorough.

### Script Name

BasicDataTypesTestSuite 

### Steps To The Task

- [x] Split up all tests within existing facts into their own facts
- [x] Add comments
- [x] Update assert descriptions

### PR DESCRIPTION
**Pull request by AdamBorek**
Facts have been broken up into multiple smaller ones

### CODE REVIEW

File: projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml [hunk 1]
```diff
@@ -2,699 +2,1325 @@
...
 		input = pointer_invalid;
 		output = bool(input);
-		assert_true(output, "#2 bool ( pointer_invalid:local ), should be true");
-		assert_typeof(output, "bool", "#2.1 bool ( pointer_invalid:local ), should return bool type");
+		assert_true(output, "bool ( pointer_invalid:local ), should be true");
+		assert_typeof(output, "bool", "bool ( pointer_invalid:local ), should return bool type");
+		
+	});
+	
+	addFact("bool_test #2", function() {
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml]:
```suggestion
	addFact("bool_test #3", function() {
```

File: projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml [hunk 2]
```diff
@@ -2,699 +2,1325 @@
...
 		input = pointer_null;
 		output = bool(input);
-		assert_false(output, "#3 bool ( pointer_null:local ), should be false");
-		assert_typeof(output, "bool", "#3.1 bool ( pointer_null:local ), should be of type bool");
+		assert_false(output, "bool ( pointer_null:local ), should be false");
+		assert_typeof(output, "bool", "bool ( pointer_null:local ), should be of type bool");
+		
+	});
+	
+	addFact("bool_test #3", function() {
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml]:
```suggestion
	addFact("bool_test #4", function() {
```

File: projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml [hunk 3]
```diff
@@ -2,699 +2,1325 @@
...
 		input = 32.33;
 		output = bool(input);
-		assert_true(output, "#4 bool ( real:local ), should be true");
-		assert_typeof(output, "bool", "#4.1 bool ( real:local ), should return bool type");
+		assert_true(output, "bool ( real:local ), should be true");
+		assert_typeof(output, "bool", "bool ( real:local ), should return bool type");
+		
+	});
+	
+	addFact("bool_test #4", function() {
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml]:
```suggestion
	addFact("bool_test #5", function() {
```

File: projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml [hunk 4]
```diff
@@ -2,699 +2,1325 @@
...
 		input = 0.5;
 		output = bool(input);
-		assert_false(output, "#5 bool ( real:local ), <= 0.5 should be false");
-		assert_typeof(output, "bool", "#5.1 bool ( real:local ), should return bool type");
+		assert_false(output, "bool ( real:local ), <= 0.5 should be false");
+		assert_typeof(output, "bool", "bool ( real:local ), should return bool type");	
+		
+	});
+	
+	addFact("bool_test #5", function() {
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml]:
```suggestion
	addFact("bool_test #6", function() {
```

File: projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml [hunk 5]
```diff
@@ -2,699 +2,1325 @@
...
 		input = 0.55;
 		output = bool(input);
-		assert_true(output, "#6 bool ( real:local ), > 0.5 should be true");
-		assert_typeof(output, "bool", "#6.1 bool ( real:local ), should return bool type");
+		assert_true(output, "bool ( real:local ), > 0.5 should be true");
+		assert_typeof(output, "bool", "bool ( real:local ), should return bool type");
+		
+	});
+	
+	addFact("bool_test #6", function() {
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml]:
```suggestion
	addFact("bool_test #7", function() {
```

File: projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml [hunk 6]
```diff
@@ -2,699 +2,1325 @@
...
 		input = int32(12);
 		output = bool(input);
-		assert_true(output, "#7 bool ( int32:local ), should be true");
-		assert_typeof(output, "bool", "#7.1 bool ( int32:local ), should return bool type");
+		assert_true(output, "bool ( int32:local ), should be true");
+		assert_typeof(output, "bool", "bool ( int32:local ), should return bool type");
+		
+	});
+	
+	addFact("bool_test #7", function() {
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml]:
```suggestion
	addFact("bool_test #8", function() {
```

File: projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml [hunk 7]
```diff
@@ -2,699 +2,1325 @@
...
 		input = int64(32);
 		output = bool(input);
-		assert_true(output, "#8 bool ( int64:local ), should be true");
-		assert_typeof(output, "bool", "#8.1 bool ( int64:local ), should return bool type");
+		assert_true(output, "bool ( int64:local ), should be true");
+		assert_typeof(output, "bool", "bool ( int64:local ), should return bool type");
+		
+	});
+	
+	addFact("bool_test #8", function() {
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml]:
```suggestion
	addFact("bool_test #9", function() {
```

File: projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml [hunk 8]
```diff
@@ -2,699 +2,1325 @@
...
 		input = "true";
 		output = bool(input);
-		assert_true(output, "#9 bool ( 'true':local ), should be true");
-		assert_typeof(output, "bool", "#9.1 bool ( 'true':local ), should return bool type");
+		assert_true(output, "bool ( 'true':local ), should be true");
+		assert_typeof(output, "bool", "bool ( 'true':local ), should return bool type");
+		
+	});
+	
+	addFact("bool_test #9", function() {
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml]:
```suggestion
	addFact("bool_test #10", function() {
```

File: projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml [hunk 9]
```diff
@@ -2,699 +2,1325 @@
...
 		input = "false";
 		output = bool(input);
-		assert_false(output, "#10 bool ( 'false':local ), should be false");
-		assert_typeof(output, "bool", "#10.1 bool ( 'false':local ), should return bool type");
+		assert_false(output, "bool ( 'false':local ), should be false");
+		assert_typeof(output, "bool", "bool ( 'false':local ), should return bool type");
+		
+	});
+	
+	addFact("bool_test #10", function() {
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml]:
```suggestion
	addFact("bool_test #11", function() {
```

File: projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml [hunk 10]
```diff
@@ -2,699 +2,1325 @@
...
 		input = true;
 		output = bool(input);
-		assert_true(output, "#11 bool ( bool - true:local ), should be true");
-		assert_typeof(output, "bool", "#11.1 bool ( bool - true:local ), should return bool type");
+		assert_true(output, "bool ( bool - true:local ), should be true");
+		assert_typeof(output, "bool", "bool ( bool - true:local ), should return bool type");
+		
+	});
+	
+	addFact("bool_test #11", function() {
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml]:
```suggestion
	addFact("bool_test #12", function() {
```

File: projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml [hunk 11]
```diff
@@ -2,699 +2,1325 @@
...
 		input = false;
 		output	= bool(input);
-		assert_false(output, "#12 bool ( bool - false:local ), should be false" );
-		assert_typeof(output, "bool", "#12.1 bool ( bool - true:local ), should return bool type");
+		assert_false(output, "bool ( bool - false:local ), should be false" );
+		assert_typeof(output, "bool", "bool ( bool - true:local ), should return bool type");
+		
+	});
+	
+	addFact("bool_test #12", function() {
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml]:
```suggestion
	addFact("bool_test #13", function() {
```

File: projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml [hunk 12]
```diff
@@ -2,699 +2,1325 @@
...
 		input = RainbowColors.Red;
 		output = bool(input);
-		assert_false(output, "#13 bool (enum.0:local), should be false");
-		assert_typeof(output, "bool", "#13.1 bool (enum.0:local), should return bool type");
+		assert_false(output, "bool (enum.0:local), should be false");
+		assert_typeof(output, "bool", "bool (enum.0:local), should return bool type");
+		
+	});
+	
+	addFact("bool_test #13", function() {
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml]:
```suggestion
	addFact("bool_test #14", function() {
```

File: projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml [hunk 13]
```diff
@@ -2,699 +2,1325 @@
...
 		input = RainbowColors.Orange;
 		output = bool(input);
-		assert_true(output, "#14 bool (enum.1:local), should be true");
-		assert_typeof(output, "bool", "#14.1 bool (enum.1:local), should return bool type");
+		assert_true(output, "bool (enum.1:local), should be true");
+		assert_typeof(output, "bool", "bool (enum.1:local), should return bool type");
+		
+	});
+	
+	addFact("bool_test #14", function() {
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml]:
```suggestion
	addFact("bool_test #15", function() {
```

File: projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml [hunk 14]
```diff
@@ -2,699 +2,1325 @@
...
 		input = NaN;
 		output = bool(input);
-		assert_false(output, "#15 bool ( nan:local ), should be false");
-		assert_typeof(output, "bool", "#15.1 bool ( nan:local ), should return bool type");
+		assert_false(output, "bool ( nan:local ), should be false");
+		assert_typeof(output, "bool", "bool ( nan:local ), should return bool type");
+		
+	});
+	
+	addFact("bool_test #15", function() {
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml]:
```suggestion
	addFact("bool_test #16", function() {
```

File: projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml [hunk 15]
```diff
@@ -2,699 +2,1325 @@
...
 		input = infinity;
 		output = bool(input);
-		assert_true(output, "#16 bool ( infinity:local ), should be true");
-		assert_typeof(output, "bool", "#16.1 bool ( infinity:local ), should return bool type");
+		assert_true(output, "bool ( infinity:local ), should be true");
+		assert_typeof(output, "bool", "bool ( infinity:local ), should return bool type");
+		
+	});
+	
+	addFact("bool_test #16", function() {
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml]:
```suggestion
	addFact("bool_test #17", function() {
```

File: projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml [hunk 16]
```diff
@@ -2,699 +2,1325 @@
...
 		input = undefined;
 		output = bool(input);
-		assert_false(output, "#17 bool ( undefined:local ), should be false");
-		assert_typeof(output, "bool", "#17.1 bool ( undefined:local ), should return bool type");
+		assert_false(output, "bool ( undefined:local ), should be false");
+		assert_typeof(output, "bool", "bool ( undefined:local ), should return bool type");
+		
+	});
+	
+	addFact("bool_test #17", function() {
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml]:
```suggestion
	addFact("bool_test #18", function() {
```

File: projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml [hunk 17]
```diff
@@ -2,699 +2,1325 @@
...
 		input = {};
 		output = bool(input);
-		assert_true(output, "#18 bool ( struct:local ), should be true");
-		assert_typeof(output, "bool", "#18.1 bool ( struct:local ), should return bool type");
+		assert_true(output, "bool ( struct:local ), should be true");
+		assert_typeof(output, "bool", "bool ( struct:local ), should return bool type");
+		
+	});
+	
+	addFact("bool_test #18", function() {
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml]:
```suggestion
	addFact("bool_test #19", function() {
```

File: projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml [hunk 18]
```diff
@@ -2,699 +2,1325 @@
...
 		input = function() {};
 		output = bool(input);
-		assert_true(output, "#19 bool ( method:local ), should be true");
-		assert_typeof(output, "bool", "#19.1 bool ( method:local ), should return bool type");
+		assert_true(output, "bool ( method:local ), should be true");
+		assert_typeof(output, "bool", "bool ( method:local ), should return bool type");
+		
+	});
+	
+	addFact("bool_test #19", function() {
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml]:
```suggestion
	addFact("bool_test #20", function() {
```

File: projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml [hunk 19]
```diff
@@ -2,699 +2,1325 @@
...
 		//#20 bool(array) should throw error
 		assert_throw(function() {
 			var input = [];
 			return bool(input);
-		}, "#20 bool ( array ), should throw error");
+		}, "bool ( array ), should throw error");
+		
+	});
+	
+	addFact("bool_test #20", function() {
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml]:
```suggestion
	addFact("bool_test #21", function() {
```

File: projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml [hunk 20]
```diff
@@ -2,699 +2,1325 @@
...
 		//#21 bool (string), where string is not 'true'/'false', should throw error
 		assert_throw(function() {
 			var input = "string";
 			return bool(input);
-		}, "#21 bool ( string ), where string is not 'true'/'false', should throw error");
+		}, "bool ( string ), where string is not 'true'/'false', should throw error");
+		
+	});
+	
+	addFact("bool_test #21", function() {
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml]:
```suggestion
	addFact("bool_test #22", function() {
```

File: projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml [hunk 21]
```diff
@@ -2,699 +2,1325 @@
...
 		//#22 bool (empty string) should throw error
 		assert_throw(function() {
 			var input = "";
 			return bool(input);
-		}, "#22 bool (empty string), should throw error");
+		}, "bool (empty string), should throw error");
+		
+	});
+
+	addFact("bool_test #22", function() {
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml]:
```suggestion
	addFact("bool_test #23", function() {
```

File: projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml [hunk 22]
```diff
@@ -2,699 +2,1325 @@
...
+		
+	});
+	
+	addFact("ptr_test #19", function() {
+		
+		var input, output;
 		
 		//#19 ptr ( undefined ), should throw error
 		assert_throw(function() {
 			var input = undefined
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml]:
```suggestion
			var input = undefined;
```

File: projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml [hunk 23]
```diff
@@ -2,699 +2,1325 @@
...
+		
+	});
+	
+	addFact("real_test #19", function() {
+			
+		var input, output;
 		
 		//#19 real ( undefined ), should throw error
 		assert_throw(function() {
 			var input = undefined
```
Comment (Indibulous) [inline projects/xUnit/scripts/BasicDataTypesTestSuite/BasicDataTypesTestSuite.gml]:
```suggestion
			var input = undefined;
```

**Review by Indibulous — APPROVED** [review]
Seems good! Just fixed the numbering for the `"bool_test"`s (both tests 2 and 3 were called `"bool test #2"`, and as a result all tests after that had an offset number) and added a couple semicolons 👍

**Review by DiasFranciscoA — APPROVED** [review]

**Pull request merged**